### PR TITLE
Add checkpoint after to_dense_mt for trio stats

### PR DESCRIPTION
**Pull request by KoalaQin**

### CODE REVIEW

File: gnomad_qc/v4/annotations/generate_variant_qc_annotations.py [hunk 1]
```diff
@@ -481,6 +481,7 @@ def run_generate_trio_stats(
     rmt = rmt.filter_cols(hl.is_defined(vmt.cols()[rmt.col_key]))
 
     mt = hl.vds.to_dense_mt(hl.vds.VariantDataset(rmt, vmt))
+    mt = mt.checkpoint(hl.utils.new_temp_file("trio_stats_dense", "mt"), overwrite=True)
```
Comment (klaricch) [inline gnomad_qc/v4/annotations/generate_variant_qc_annotations.py:484]:
```suggestion
    mt = mt.checkpoint(hl.utils.new_temp_file("trio_stats_dense", "mt"))
```

Comment (klaricch) [reply gnomad_qc/v4/annotations/generate_variant_qc_annotations.py:484]: the overwrite arg isn't needed when using new_temp_file

Comment (KoalaQin) [reply gnomad_qc/v4/annotations/generate_variant_qc_annotations.py:484]: L560, L621, L664 had overwrite too, should I remove them?

Comment (klaricch) [reply gnomad_qc/v4/annotations/generate_variant_qc_annotations.py]: yeah those can be removed too

**Review by klaricch — APPROVED** [review]
lgtm

**Pull request merged**